File: Assets/zfoocs/Common/TripleLSS.cs
using System;
using System.Collections.Generic;

namespace zfoocs
{
    
    public class TripleLSS
    {
        public long left;
        public string middle;
        public string right;

        public static TripleLSS ValueOf(long left, string middle, string right)
        {
            var packet = new TripleLSS();
            packet.left = left;
            packet.middle = middle;
            packet.right = right;
            return packet;
        }
    }


    public class TripleLSSRegistration : IProtocolRegistration
    {
        public short ProtocolId()
        {
            return 116;
        }

        public void Write(ByteBuffer buffer, object packet)
        {
            if (packet == null)
            {
                buffer.WriteInt(0);
                return;
            }
            TripleLSS message = (TripleLSS) packet;
            buffer.WriteInt(-1);
            buffer.WriteLong(message.left);
            buffer.WriteString(message.middle);
            buffer.WriteString(message.right);
        }

        public object Read(ByteBuffer buffer)
        {
            int length = buffer.ReadInt();
            if (length == 0)
            {
                return null;
            }
            int beforeReadIndex = buffer.ReadOffset();
            TripleLSS packet = new TripleLSS();
            long result0 = buffer.ReadLong();
            packet.left = result0;
            string result1 = buffer.ReadString();
            packet.middle = result1;
            string result2 = buffer.ReadString();
            packet.right = result2;
            if (length > 0) {
                buffer.SetReadOffset(beforeReadIndex + length);
            }
            return packet;
        }
    }
}

File: Assets/zfoolua/common/TripleLSS.lua

local TripleLSS = {}

function TripleLSS:new()
    local obj = {
        left = 0, -- long
        middle = "", -- string
        right = "" -- string
    }
    setmetatable(obj, self)
    self.__index = self
    return obj
end

function TripleLSS:protocolId()
    return 116
end

function TripleLSS:write(buffer, packet)
    if packet == nil then
        buffer:writeInt(0)
        return
    end
    buffer:writeInt(-1)
    buffer:writeLong(packet.left)
    buffer:writeString(packet.middle)
    buffer:writeString(packet.right)
end

function TripleLSS:read(buffer)
    local length = buffer:readInt()
    if length == 0 then
        return nil
    end
    local beforeReadIndex = buffer:getReadOffset()
    local packet = TripleLSS:new()
    local result0 = buffer:readLong()
    packet.left = result0
    local result1 = buffer:readString()
    packet.middle = result1
    local result2 = buffer:readString()
    packet.right = result2
    if length > 0 then
        buffer:setReadOffset(beforeReadIndex + length)
    end
    return packet
end

return TripleLSS
